src/aws-cpp-sdk-core/source/Aws.cpp

```diff
@@ -26,8 +26,20 @@ namespace Aws
 {
     static const char* ALLOCATION_TAG = "Aws_Init_Cleanup";
 
+    static std::mutex s_initShutdownMutex;
+    static size_t s_initCount = 0;
+
     void InitAPI(const SDKOptions &options)
     {
+        std::unique_lock<std::mutex> lock(s_initShutdownMutex);
+        s_initCount += 1;
+        if(s_initCount != 1)
```

Comment:
could we use an atomic int instead of a mutex?

Reply:
I also wanted to prevent simultaneous Init/Shutdown calls.